https/tls_config.go

```diff
@@ -0,0 +1,112 @@
+// Package https allows the implementation of tls
+package https
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/prometheus/common/log"
+	"gopkg.in/yaml.v2"
+)
+
+type Config struct {
+	TLSCertPath string    `yaml:"tlsCertPath"`
+	TLSKeyPath  string    `yaml:"tlsKeyPath"`
+	TLSConfig   TLSStruct `yaml:"tlsConfig"`
+}
+
+type TLSStruct struct {
+	RootCAs            string `yaml:"rootCAs"`
+	ServerName         string `yaml:"serverName"`
+	ClientAuth         string `yaml:"clientAuth"`
+	ClientCAs          string `yaml:"clientCAs"`
+	InsecureSkipVerify bool   `yaml:"insecureSkipVerify"`
```

Comment:
Do we need this parameter for a TLS server?

Reply:
I added it as it's useful for testing and also an option in prometheus

Reply:
From the documentation:

> InsecureSkipVerify controls whether a client verifies the server's certificate chain and host name.

This is useless for a server. Prometheus has it because it acts as a TLS client.

Reply:
See comment above. Would be needed for securing Alertmanager cluster traffic.

Reply:
Talking about this with @simonpasquier offline. We can remove it here for now and add it again later once we want to reuse it in Alertmanager.